packages/virtual/src/index.tsx

```diff
@@ -1,71 +1,123 @@
 import { For, createSignal } from "solid-js";
 import type { Accessor, JSX } from "solid-js";
+import type { DOMElement } from "solid-js/jsx-runtime";
+import { access } from "@solid-primitives/utils";
+import type { MaybeAccessor } from "@solid-primitives/utils";
+
+type VirtualListConfig<T extends readonly any[]> = {
+  items: MaybeAccessor<T | undefined | null | false>;
+  rootHeight: MaybeAccessor<number>;
+  rowHeight: MaybeAccessor<number>;
+  overscanCount?: MaybeAccessor<number>;
+};
+
+type VirtualListReturn<T extends readonly any[]> = [
+  {
+    containerHeight: Accessor<number>;
+    viewerTop: Accessor<number>;
+    visibleItems: Accessor<T>;
+  },
```

Comment:
```suggestion
  Accessor<{
    containerHeight: number;
    viewerTop: number;
    visibleItems: T;
  }>,
```
Since this is all derived from a single signal, I would prefer this to be a single `Accessor`. Each `Accessor` suggests that there is a signal specifically to that property.
So for example, if you wrote:
```ts
const [virtual, updateVirtual] = createVirtualList(...)
createEffect(() => {
	let h = virtual.containerHeight()
	console.log("height:", h)
})
```
Would you expect this to rerun even though height didn't actually change? Just because internal `offset` signal got triggered.
To have `Accessor` for each property, or to make them getters—which is kinda the same thing, each property would have to be memoized to only rerun when it changes. But it's probably not worth it.

Reply:
ah, I see, that makes sense